helix-term/src/commands.rs

```diff
@@ -1446,7 +1536,10 @@ fn half_page_down(cx: &mut Context) {
     scroll(cx, offset, Direction::Forward);
 }
 
+#[allow(deprecated)]
 fn copy_selection_on_line(cx: &mut Context, direction: Direction) {
+    use helix_core::{pos_at_visual_coords, visual_coords_at_pos};
```

Comment:
Yeah right now `copy_selection_on_line` and `align_selection` use the old positioning functions. I marked these as deprected because in 95% of cases you should be using the new positoning functions.

However in these two cases it made sense (to me) to ignore (inline) virtual text and softwrap. In that case using the old functions is correct. 
Addings a versionof atleast `copy_selection_on_line` that uses visual lines in the future might make sense. I sort of held off adding too many of these commands as there are lots of commands that use lines (`x` or `X` do too for example) and I wanted to avoid making this PR too large. These commands should be easy to add in followup PRs.